slang.h

```diff
@@ -3950,11 +3955,23 @@ namespace slang
             LayoutRules     rules = LayoutRules::Default,
             ISlangBlob**    outDiagnostics = nullptr) = 0;
 
-        virtual SLANG_NO_THROW TypeLayoutReflection* SLANG_MCALL getParameterBlockLayout(
+            /** Get a container type from `elementType`. For example, given type `T`, returns
+                a type that represents `StructuredBuffer<T>`.
+
+                @param `elementType`: the element type to wrap around.
+                @param `containerType`: the type of the container to wrap `elementType` in.
+                @param `outDiagnostics`: a blob to receive diagnostic messages.
+            */
+        virtual SLANG_NO_THROW TypeReflection* SLANG_MCALL getContainerType(
             TypeReflection* elementType,
-            SlangInt        targetIndex = 0,
-            LayoutRules     rules = LayoutRules::Default,
-            ISlangBlob**    outDiagnostics = nullptr) = 0;
+            ContainerType containerType,
+            ISlangBlob** outDiagnostics = nullptr) = 0;
+
+            /** Return a `TypeReflection` that represents the `__Dynamic` type.
+                This type can be used as a specialization argument to indicate using
+                dynamic dispatch.
+            */
+        virtual SLANG_NO_THROW TypeReflection* SLANG_MCALL getDynamicType() = 0;
 
             /** Get the mangled name for a type RTTI object.
             */
```

Comment:
A minor note here is that when possible we should add any new `virtual` methods to the *end* of an interface declaration, so that it preserves binary compatibility with client code compiled against the old declaration.